phabfive/maniphest.py

```diff
@@ -49,115 +51,247 @@ def create_from_config(self, config_file, dry_run=False):
-                return (False, None)
-
-        #####
-        # STEP: Render
-        log.debug(data)
-
-        # Define the variables to be used in the template
-        variables = data["variables"]
-
-        rendered_data = []
-
-        for ticket_data in data["tickets"]:
-            rendered_data.append({
-                "title": Template(ticket_data["title"]).render(variables),
-                "description": Template(ticket_data["description"]).render(variables),
-                "projects": ticket_data["projects"],
-            })
-
-        log.debug(rendered_data)
-
-        #####
-        # STEP: Post render data validation
-        for ticket_data in rendered_data:
-            if not isinstance(ticket_data["title"], str):
-                log.error("Ticket title must be a string")
-
-            if not isinstance(ticket_data["description"], str):
-                log.error("Ticket description must be a string")
-
-        #####
-        # STEP: Run
-        if dry_run:
-            log.critical("Running with --dry-run, tickets will NOT be commited to phabricator")
-
-        result_ids = []
-
-        for ticket_data in rendered_data:
-            # Prepare the data to submit for each ticket
-            transactions = [
-                {"type": "title", "value": ticket_data["title"]},
-                {"type": "description", "value": ticket_data["description"]},
-                {"type": "priority", "value": ticket_data.get("priority", TICKET_PRIORITY_NORMAL)},
-            ]
-
-            if ticket_data.get("projects"):
-                transactions.append({
-                    "type": "projects.set",
-                    "value": [
-                        project_name_to_id_mapping[project_name]
-                        for project_name in ticket_data["projects"]
-                    ],
-                })
-
-            log.debug("transactions for ticket")
-            log.debug(transactions)
-
-            if not dry_run:
-                result = self.phab.maniphest.edit(
-                    transactions=transactions,
-                )
-
-                result_ids.append(result["object"]["id"])
-
-                log.debug("ticket create result")
-                log.debug(result)
-
-            log.info("Created ticket")
-
-        #####
-        # STEP: Report
-        if dry_run:
-            log.critical("Running with --dry-run, no ticket details will be reported as nothing was created")
-            return (True, None)
-
-        log.info("Report of all created tickets")
-        for result_id in result_ids:
-            _, ticket_info = self.info(int(result_id))
-            log.info(f"Title: {ticket_info['title']} | ID: {ticket_info['id']} | URI: {ticket_info['uri']}")
+                log.critical(f"Hard errors found during pre-processing step. Please update config or phabricator server and try again. Nothing has been commited yet.")
+                sys.exit(1)
```

Comment:
Use `sys.exit()` only in the CLI not in the library code. Raise an exception instead.